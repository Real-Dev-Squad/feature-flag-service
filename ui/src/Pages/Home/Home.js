import Dashboard from "../../Components/Dashboard/Dashboard";
import Layout from "../../Components/Layout/Layout";
import { FEATURE_FLAGS, FEATURE_FLAGS_HEADERS} from "../../Mock/featureFlags";

export default function Home(){
    return(
        <Layout>
            <Dashboard 
            featureFlags={FEATURE_FLAGS} 
            headers={FEATURE_FLAGS_HEADERS}
            />
      </Layout>
    )
}



