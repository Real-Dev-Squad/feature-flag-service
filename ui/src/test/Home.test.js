import { render } from '@testing-library/react';
import Home from '../Pages/Home/Home';

describe('<Home/> component', () => {
    it('component should render',() => {
        render(<Home />);
    });
})