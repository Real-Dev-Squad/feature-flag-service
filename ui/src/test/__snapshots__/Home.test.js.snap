// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> component should match snapshot 1`] = `
<div>
  <header
    class="py-2 px-8 h-20 flex items-center bg-rds-primary text-slate-100"
  >
    <img
      class="rounded-full h-12 block mx-2"
      src="https://res.cloudinary.com/realdevsquad/image/upload/v1649837643/profile/EYNHBSK7riBKY775OOMB/mfzbgspqgvj6unz34unm.jpg"
    />
    <h1
      class="text-lg"
    >
      Pallab Sonowal
    </h1>
    <button
      class="py-2 px-4 rounded-lg ml-auto bg-white text-rds-primary hover:bg-slate-400"
    >
      Logout
    </button>
  </header>
  <main
    class="flex flex-row my-4"
  >
    <nav
      class="px-3"
    >
      <ul>
        <li
          class="text-blue-600 hover:underline text-sm py-1"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="text-blue-600 hover:underline text-sm py-1"
        >
          <a
            class=""
            href="/users"
          >
            Create Users
          </a>
        </li>
        <li
          class="text-blue-600 hover:underline text-sm py-1"
        >
          <a
            class=""
            href="/flags"
          >
            Flags
          </a>
        </li>
        <li
          class="text-blue-600 hover:underline text-sm py-1"
        >
          <a
            class=""
            href="/SDK"
          >
            SDK
          </a>
        </li>
        <li
          class="text-blue-600 hover:underline text-sm py-1"
        >
          <a
            class=""
            href="/docs"
          >
            Docs
          </a>
        </li>
      </ul>
    </nav>
    <section>
      <table
        class="shadow text-sm text-left table-fixed w-3/4 max-w-screen-md"
      >
        <thead>
          <tr
            class="text-teal-700 text-xs"
          >
            <th>
              <tr>
                <th
                  class="p-4"
                >
                  Feature Flag
                </th>
                <th />
              </tr>
            </th>
            <th>
              <tr>
                <th
                  class="p-4"
                >
                  Repository
                </th>
                <th>
                  <div>
                    <div
                      class="up-arrow null"
                      title="ascending"
                    />
                    <div
                      class="down-arrow null"
                      title="descending"
                    />
                  </div>
                </th>
              </tr>
            </th>
            <th>
              <tr>
                <th
                  class="p-4"
                >
                  Created At
                </th>
                <th>
                  <div>
                    <div
                      class="up-arrow null"
                      title="ascending"
                    />
                    <div
                      class="down-arrow null"
                      title="descending"
                    />
                  </div>
                </th>
              </tr>
            </th>
            <th
              class="p-4"
            >
              Enabled
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-t"
          >
            <td
              class="px-4 py-2"
            >
              <span>
                Dark Mode
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <span>
                Member Site
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <span>
                1657868257983
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <label
                class="w-14 h-7 rounded-3xl block relative p-1 ease-in-out duration-300 bg-sky-500"
              >
                <input
                  checked=""
                  class="hidden"
                  type="checkbox"
                />
                <span
                  class="w-6 h-full block rounded-3xl bg-white relative ease-in-out duration-300 false"
                />
              </label>
            </td>
          </tr>
          <tr
            class="border-t"
          >
            <td
              class="px-4 py-2"
            >
              <span>
                Cloudinary
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <span>
                Welcome Site
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <span>
                1657868057983
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <label
                class="w-14 h-7 rounded-3xl block relative p-1 ease-in-out duration-300 bg-slate-300"
              >
                <input
                  class="hidden"
                  type="checkbox"
                />
                <span
                  class="w-6 h-full block rounded-3xl bg-white relative ease-in-out duration-300 translate-x-full"
                />
              </label>
            </td>
          </tr>
          <tr
            class="border-t"
          >
            <td
              class="px-4 py-2"
            >
              <span>
                Get Intro
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <span>
                Member Site
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <span>
                1657868240983
              </span>
            </td>
            <td
              class="px-4 py-2"
            >
              <label
                class="w-14 h-7 rounded-3xl block relative p-1 ease-in-out duration-300 bg-sky-500"
              >
                <input
                  checked=""
                  class="hidden"
                  type="checkbox"
                />
                <span
                  class="w-6 h-full block rounded-3xl bg-white relative ease-in-out duration-300 false"
                />
              </label>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </main>
  <footer
    class="my-8 w-screen flex justify-center text-xs"
  >
    <span
      class="ml-2"
    >
      Copyright 2022-2023
    </span>
    <a
      class="ml-2 hover:underline"
      href="http://www.google.com"
    >
      RDS Website
    </a>
    <a
      class="ml-2 hover:underline"
      href="http://www.google.com"
    >
      Discord server
    </a>
  </footer>
</div>
`;
